lib/containers/utils.pm

```diff
@@ -315,4 +315,96 @@ sub reset_container_network_if_needed {
     }
 }
 
+sub test_volumes {
+    my %args = @_;
+    my $runtime = $args{runtime};
+
+    # From https://docs.docker.com/storage/bind-mounts/
+    # The --mount flag does not support z or Z options for modifying selinux labels.
+    my $Z = $runtime eq "podman" ? ",Z" : "";
+
+    my $test_dir = "test_dir";
+    my $test_file = "test_file";
+    my $test_image = "test_image";
+    my $test_container = "test_container";
+    my $test_volume = "test_volume";
+
+    # Create a directory with the sticky-bit set because of rootless podman
+    assert_script_run("mkdir -m 1777 $test_dir");
+
+    # Create Dockerfile with VOLUME defined
+    assert_script_run("echo -e 'FROM busybox\nVOLUME /$test_dir' > $test_dir/Dockerfile");
+
+    # Build image
+    assert_script_run("$runtime build -t $test_image -f $test_dir/Dockerfile $test_dir/");
+
+    assert_script_run("touch $test_dir/$test_file");
+
+    # Test --volumes-from
+    assert_script_run("$runtime run -d --name $test_container $test_image");
+    assert_script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir | grep $test_file");
+    assert_script_run("$runtime rm -vf $test_container");
+
+    assert_script_run("$runtime run -d --name $test_container -v \$PWD/$test_dir:/$test_dir:Z $test_image");
```

Comment:
Also, should this be executed also on Docker engine? If not, I would exclude Docker from this or move this to the end and return earlier in case of Docker.

Reply:
> Also, should this be executed also on Docker engine? If not, I would exclude Docker from this or move this to the end and return earlier in case of Docker.

Also Docker, yes.  It was ok on my Bash test code.